### PR TITLE
"Solves 1719"

### LINKED ISSUE #1719: Accessing private endpoints should redirect to home page.
**I'm submitting a ...**
- [x] bug report
- [ ] feature request

**Current behaviour:**

Accessing endpoints like `http://badgeyay.com/my-badges`, if one is not logged in, shows loading and then a blank page. 
**Expected behaviour:**

Instead we should redirect the user to home page.
**Steps to reproduce:**
Go to [badgeyay](http://badgeyay.com/), and logout if you are currently logged in and then go to [my-badges](http://badgeyay.com/my-badges) page.

---
_Working on it_

### PR DESCRIPTION
**Pull request by sk9331657**
Fixes #1719 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

-This will route create-badges to login page in case user is not logged in.
ezgif com-video-to-gif 4

-

### CODE REVIEW

**Comment by sk9331657** [comment]
@ParthS007   @abishekvashok  Please review.

**Comment by abishekvashok** [comment]
Please:
- check the build logs and rectify the build
- (optional): on such redirects display a label on the login page like: "please login to continue to xyz" or simply "please login to continue"

Cheers

**Comment by sk9331657** [comment]
@gabru-md  @ParthS007 This can be now merged. :smile:

File: frontend/app/routes/create-badges.js [hunk 1]
```diff
@@ -4,13 +4,14 @@ import Route from '@ember/routing/route';
 const { RSVP, set } = Ember;
 
 export default Route.extend({
+  beforeModel() {
+    if (this.get('session.currentUser') == undefined) {
```
Comment (ParthS007) [inline frontend/app/routes/create-badges.js]: Use Strict equality here.

File: frontend/app/routes/create-badges.js [hunk 2]
```diff
@@ -4,13 +4,14 @@ import Route from '@ember/routing/route';
...
+      this.transitionTo('login');
+    }
+  },
   model() {
-    return RSVP.hash({
-      def_images : this.get('store').findAll('def-image'),
-      user       : this.get('store').findRecord('user', this.get('session.currentUser').uid)
-    });
+    return RSVP.hash({ 'def_images': this.get('store').findAll('def-image'), 'user': this.get('store').findRecord('user', this.get('session.currentUser').uid) });
```
[the remark this answers is not in the document]
Comment (sk9331657) [reply frontend/app/routes/create-badges.js]: @ParthS007  Travis CI is throwing build error with previous code.

**Comment by sk9331657** [comment]
@ParthS007 Now finally Travis CI has passed the code.Please merge it . :smile:

**Pull request closed without merging**